cord19-explorer/src/main/java/edu/asu/diging/cord19/explorer/core/service/worker/impl/CoordinateCleanerImpl.java

```diff
@@ -0,0 +1,246 @@
+import edu.asu.diging.cord19.explorer.core.model.impl.WikipediaArticleImpl;
+
+
+@Service
+@PropertySource({ "classpath:config.properties", "${appConfigFile:classpath:}/app.properties" })
+public class CoordinateCleanerImpl implements CoordinateCleaner {
+
+
+    
+    @Value("${mongo.duplicate.database.name}")
+    private String duplicateCollection;
+
+    @Autowired
+    private TaskRepository taskRepo;
+
+    @Autowired
+    private PublicationRepository pubRepo;
+
+    @Autowired
+    private MongoTemplate mongoTemplate;
+    
+    @Autowired
+    private MapTotalsRepository totalsRepo;
+ 
+    
+    
+    private double processLength3Coords(List<String> coords) {
+        float degrees = Float.parseFloat(coords.get(0));
+        float minutes = Float.parseFloat(coords.get(1));
+        float seconds = Float.parseFloat(coords.get(2));
+        double decimalDegrees = 0;
+        if(degrees == 0) {
+            decimalDegrees = ((minutes / 60.0) + (seconds / 3600.0));
+        } else {
+            decimalDegrees = Math.signum(degrees) * (Math.abs(degrees) + (minutes / 60.0) + (seconds / 3600.0));
+        }
+        return decimalDegrees;
+    }
+    
+    private double processLength2Coords(List<String> coords) {
+        float degrees = Float.parseFloat(coords.get(0));
+        float minutes = Float.parseFloat(coords.get(1));
+        double decimalDegrees = Math.signum(degrees) * (Math.abs(degrees) + (minutes / 60.0));
+        return decimalDegrees;
+    }
+    
+    private CleanedCoordinatesImpl createCleanCoords(List<Double> formattedCoords) {
+        CleanedCoordinatesImpl cleanedCoords = new CleanedCoordinatesImpl();
+        cleanedCoords.setType(CoordType.Point);
+        cleanedCoords.setCoordinates(formattedCoords);
+        return cleanedCoords;
+    }
+    
+    private void parseCoords(WikipediaArticleImpl article) {
+        List<String> coords = new ArrayList<String>();
+        List<Double> formattedCoords = new ArrayList<Double>();
+        String [] splitted =  article.getCoordinates().split("\\|");
+        // split on pipes then use Ordinals to gather Degree, Mins, Secs then compute Decimal Degrees
+        for (String s: splitted) {      
+             if(s.matches("-?\\d+(\\.\\d+)?")) {
+                 coords.add(s);
+                 
+             } else if(s.equals("N") || s.equals("E")){
+                 if(coords.size() == 3) {
+                     formattedCoords.add(processLength3Coords(coords));
+                     coords.clear();
+                 }
+                 else if(coords.size() == 2) {
+                     formattedCoords.add(processLength2Coords(coords));
+                     coords.clear();
+                 }
+                 else if(coords.size() == 1) {
+                     Double doubleCoord = Double.parseDouble(coords.get(0));
+                     formattedCoords.add(doubleCoord);
+                     coords.clear();
```

Comment:
since `clear` is called in every if, just call it after the `else if`? Also, should there be coordinates that have 4 components for some reason (error in wikipedia), right now, coords wouldn't be emptied.